### ISSUE TITLE
[LOT-4] A process can set its tickets with a system call

### ISSUE THREAD

**Issue by hatfield-c** [hatfield-c/xv6-project2#4]
A system call titled "settickets(int)" will be created. This system call receives an integer as an argument, and sets the number of tickets for the process that called it equal to the argument. It will return a 0 if successful, or a -1 otherwise.

The biggest hurdle to overcome, is determining the exact mechanism that the data contained within the process's **proc** structure can be accessed, set, and saved within the system call. Theoretically, the context does not change when the system call is instantiated, however due to the nature of a system call, it will try to interrupt its way into kernel space, which might cause some unwanted context switching to occur. If this is the case, we will need to devise a more sophisticated means of passing the reference to the number of tickets into the system call, however much more experimenting must be done before we can determine the exact nature of this.

**Comment by hatfield-c** [comment]
I have been able to set up the system call successfully, and can pass information into it as well. I can also access the ticket number of a process within the system call, but I have yet to confirm if it is the same process that utilized the system (that is, I have not confirmed persistence of data yet).

I will try outputing the PID of process that instantiates the syscall, followed by its associated ticket number as well. If these match up, then we have confirmed persistence. If not however, then we will need to examine how we can enable persistence (possibly via a pipe).
